### ISSUE TITLE
Backtesting & Pyramid Trading - Scaling into a postion

### ISSUE THREAD

**Issue by wesssup** [ta4j/ta4j#1106]
I have a question , in live trading it's easy to implement Pyramid Trading (https://www.investopedia.com/articles/trading/09/pyramid-trading.asp) also called scaling into a postion.... but I could not find it inside the backtesting ... is there a way to do it ?

**Comment by TheCookieLab** [comment]
There is not, at least not out of the box. TradingRecord only allows one
open position at a time so you’d have to hack together multiple trading
records to manage something similar. Enhancing TradingRecord to support
multiple positions is on my todo list but as I’ve been preoccupied as of
late. Perhaps Soon™

**Comment by wesssup** [comment]
Agree,and I’ve seen the  Projekt Roadmap | Ta4j - Technical Analysis for Java!<https://ta4j.github.io/ta4j-wiki/Roadmap-and-Tasks.html> But it’s not the same, multiple trades or scaling into a trade, or is due to the limited information not clear what you exact try to do…

Van: nimo23 ***@***.***>
Verzonden: zaterdag 2 september 2023 23:10
Aan: ta4j/ta4j ***@***.***>
CC: Wesley Wenselaers ***@***.***>; Author ***@***.***>
Onderwerp: Re: [ta4j/ta4j] Pyramid Trading - Scaling into a position - DCA Trading possible ? (Issue #1106)


For both, it would be good to just check for existing issues beforehand, rather than opening a new one and posting your comments, suggestions, or future plans here:

#755<https://github.com/ta4j/ta4j/issues/755>

**Comment by TheCookieLab** [comment]
@Wesw very promising - would you be willing to create a draft PR to
facilitate collaboration for this?

**Comment by TheCookieLab** [comment]
Closing as superseded by broader scaling/multi-entry discussions. This issue is effectively a specific manifestation of that larger design space, so keeping it separate fragments implementation history. Please continue in the core tracking issue(s) for position-scaling support.
